_posts/2017-01-22-Swift-at-Artsy.md

```diff
@@ -0,0 +1,245 @@
+
+We consider ourselves blocked on Android support, specifically by having an engineer in our team with _deep_ experience in Android. Moving to React Native does not obviate our native skills, you're going to be significantly better in that environment with those skills than without. As we mentioned in our announcement:
+
+> If you’re not already knowledgeable about iOS development, are not motivated to put in the time to learn about the
+  platform specific details, and think making rich iOS applications with React Native will be a breeze, you’ll
+  [come home from a very cold fair](http://www.dwotd.nl/2008/06/443-van-een-kouwe-kermis-thuiskomen.html) indeed. 
+
+We need someone with a similar depth of knowledge in the Android eco-system, but we may need one or two for the entire team. The rest can continue to be a mix of Web and iOS engineers. You gain a subset of cross-platform skills using React Native. Had we continued down the path of using Swift, our skills would continue to be siloed.
+
+There is an argument that Swift will be running servers soon, and so you can re-use Swift code across platforms. I could see myself writing server-side back-end code in Swift (you're writing systems, not apps) but it has a [long way to go][ssswift]. It also isn't an argument towards using it in our native apps, we'd have to re-write servers and implement our own GraphQL and Relay stack. 
+
+With respect to Swift on Android, potentially, logic code could be shared between platforms but realistically for our setup that's just not worth it. We're moving that kind of logic into the GraphQL instance and sharing across _all_ clients, not only native platforms. If you're sharing model code, you could generate that per-project instead from the server. Since GraphQL is strongly-typed, we're doing this for both [TypeScript + GraphQL][gql2ts] and [TypeScript + Relay][vscode-relay].
+
+#### Owning the stack
+
+Pick an abstraction level of our application above UIKit and we can fork it. All our tools can be also be forked. We can fix our own issues.
+
+There are no concepts like, _"We'll use Steipete's fork of UIKit for UIPopover rotation fixes"_ or _"My version of Xcode will run tests when you press save."_
+
+In the mobile team, we have submitted code to: React Native, Relay, VS Code, Jest, [more?] - fixing problems where we see them, offering features if we need them. Some of these changes are 
+
+- Swift is working to make fixes at UIKit level even harder with [closed objects by default][closed]. 
+
+I've never heard of a team using their own version of Swift, 
+
+
+
+* Conceptual idea of customizing your language to the project
+  - We pick and choose language features we want
+
+* "Forking" / Contributing back (can't use our own fork of Swift)
+  - You can expect a reply to an issue, you don't to a radar
+  - Relay
+  - VS Code
+  - React Native
+
+* Open but hard to be accessible
+ - You need to be a compiler engineer to improve Swift
```

Comment:
Ehhhhhh, not really but I agree that the barrier to entry is far, far higher than improving JS tools. Maybe reword?